### PR TITLE
fix(behavior_velocity_planner): sort unique_lane_ids for merge_from_private_area

### PR DESCRIPTION
**Pull request by tkimura4**
Signed-off-by: tomoya.kimura <tomoya.kimura@tier4.jp>

## Description

Sort unique_lane_ids in getLaneletsOnPath for merge_from_private_area bacause this module is assumed that 
 unique_lane_ids are in order
https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_velocity_planner/src/scene_module/intersection/manager.cpp#L118-L124

( Before the change, the stop point of merge_from_private_area was wrong. )

Before the change
image

After the change
image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kosuke55 — APPROVED** [review]
LGTM

**Comment by kosuke55** [comment]
Could you please modify the following as well?
https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_velocity_planner/include/utilization/util.hpp#L305-L337
https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_velocity_planner/src/scene_module/crosswalk/manager.cpp#L30-L56

File: planning/behavior_velocity_planner/src/utilization/util.cpp [hunk 1]
```diff
@@ -593,34 +593,62 @@ LineString2d extendLine(
...
     lanelet::utils::query::getCurrentLanelets(
       lanelet::utils::query::laneletLayer(lanelet_map), current_pose, &current_lanes) &&
     nearest_segment_idx) {
     for (const auto & ll : current_lanes) {
-      if (
-        ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0) ||
-        ll.id() == path.points.at(*nearest_segment_idx + 1).lane_ids.at(0)) {
-        unique_lane_ids.insert(ll.id());
+      if (ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0)) {
+        *nearest_lane_id = ll.id();
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/src/utilization/util.cpp:613]: @tkimura4 
Can you also fix this nearest_lane_id uninitialized pointer?
L623 too

**Review by taikitanaka3 — CHANGES_REQUESTED** [review]

File: planning/behavior_velocity_planner/src/utilization/util.cpp (hunk 1 above)
Comment (tkimura4) [reply planning/behavior_velocity_planner/src/utilization/util.cpp:617]: Fixed at [b725fd3](https://github.com/autowarefoundation/autoware.universe/pull/1323/commits/b725fd34bf97f5c213fd9a13eb171ddb8ff480ac)

Comment (taikitanaka3) [reply planning/behavior_velocity_planner/src/utilization/util.cpp:617]: you are inserting value to boost::get() method at L622
I think this line should be nearest_lane_id = ll.id();
```
#9  0x00007f74f3877c01 in behavior_velocity_planner::planning_utils::getNearestLaneId (path=..., lanelet_map=std::shared_ptr<lanelet::LaneletMap> (use count 3, weak count 0) = {...}, current_pose=..., 
    nearest_segment_idx=...) at /home/t4tanaka/workspace/humble_autoware/src/universe/autoware.universe/planning/behavior_velocity_planner/src/utilization/util.cpp:617
617	        *nearest_lane_id = ll.id();

```

File: planning/behavior_velocity_planner/src/utilization/util.cpp [hunk 2]
```diff
@@ -593,34 +593,66 @@ LineString2d extendLine(
...
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/src/utilization/util.cpp:617]:
```suggestion
        nearest_lane_id = ll.id();
```

File: planning/behavior_velocity_planner/src/utilization/util.cpp [hunk 3]
```diff
@@ -593,34 +593,66 @@ LineString2d extendLine(
...
+        return nearest_lane_id;
+      }
+    }
+
+    // if the lane_id of nearest_segment_idx does not involved in current_lanes,
+    // search the lane_id of nearest_segment_idx + 1
+    *nearest_segment_idx += 1;
+    for (const auto & ll : current_lanes) {
+      if (ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0)) {
+        *nearest_lane_id = ll.id();
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/src/utilization/util.cpp:627]:
```suggestion
        nearest_lane_id = ll.id();
```

Comment (taikitanaka3) [reply planning/behavior_velocity_planner/src/utilization/util.cpp:627]: This is going to fix planner dies below case
image

**Review by taikitanaka3 — APPROVED** [review]
LGTM
image
image

**Pull request merged**